service/storage_service.cc

```diff
@@ -365,6 +370,26 @@ void storage_service::prepare_to_join(std::vector<inet_address> loaded_endpoints
         }
     }
 
+    if (replacing_a_node_with_same_ip) {
+        slogger.info("Replacing a node with same IP address, my address={}, node being replaced={}",
+                    get_broadcast_address(), get_broadcast_address());
+        slogger.info("Update tokens for replacing node early, replacing node has the same IP address of the node being replaced");
+        // We want the replacing node and the rest of the cluster has the same view of the normal tokens.
+        _token_metadata.update_normal_tokens(_bootstrap_tokens, get_broadcast_address());
```

Comment:
Not being familiar with this code, I have to admit I couldn't figure out the full meaning of this patch. Unless I'm missing something, it seems like it's changing the behavior in this case. So does this patch have cause any functional changes? Does it fix any bug? And how is this even relevant to making the boot process faster?